tests/cspc/provisioning/provisioning_test.go

```diff
@@ -0,0 +1,146 @@
+	ProvisioningTest("stripe", 1)
+	ProvisioningTest("mirror", 2)
+	ProvisioningTest("raidz", 5)
+	ProvisioningTest("raidz2", 6)
+})
+
+func ProvisioningTest(poolType string, bdCount int) {
+	var cspc *cstor.CStorPoolCluster
+	Describe("Provisioning and cleanup of CSPC", func() {
+		Context("Creating cspc", func() {
+			Specify("no error should be returned", func() {
+				cspc = cspcspecbuilder.
+					NewCSPCSpecBuilder(cspcsuite.CSPCCache, cspcsuite.infra).
+					BuildCSPC(poolType, bdCount).
+					GetCSPCSpec()
+				_, err := cspcsuite.
+					client.
+					OpenEBSClientSet.
+					CstorV1().
+					CStorPoolClusters(cspc.Namespace).
+					Create(cspc)
+				Expect(err).To(BeNil())
+			})
+
+			Specify("desired count should be on cspc",
+				func() {
+					gotCount := cspcsuite.
+						client.
+						GetDesiredInstancesStatusOnCSPC(cspc.Name, cspc.Namespace, int32(cspcsuite.infra.NodeCount))
+					Expect(gotCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+				})
+
+		})
+
+		Context("All the cspi(s) of the cspc", func() {
+			It("Should be healthy", func() {
+				gotHealthyCSPiCount := cspcsuite.
+					client.
+					GetOnlineCSPICountEventually(cspc.Name, cspc.Namespace, cspcsuite.infra.NodeCount)
+				Expect(gotHealthyCSPiCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+			})
+		})
+
+		Context("Staus of the cspc i.e. provisionedInstances and healthyInstances ", func() {
+			It("Should be updated", func() {
+				gotProvisionedCount := cspcsuite.
+					client.
+					GetProvisionedInstancesStatusOnCSPC(cspc.Name, cspc.Namespace, int32(cspcsuite.infra.NodeCount))
+				Expect(gotProvisionedCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+
+				gotHealthyCount := cspcsuite.
+					client.
+					GetHealthyInstancesStatusOnCSPC(cspc.Name, cspc.Namespace, int32(cspcsuite.infra.NodeCount))
+				Expect(gotHealthyCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+			})
+		})
+
+		Context("Deleting the cspc", func() {
+
+			It("No error should be returned", func() {
+				err := cspcsuite.
+					client.
+					OpenEBSClientSet.
+					CstorV1().
+					CStorPoolClusters(cspc.Namespace).
+					Delete(cspc.Name, &metav1.DeleteOptions{})
+				Expect(err).To(BeNil())
+			})
+
+			It("No corresponding cspi(s) should be present", func() {
+				gotCSPICount := cspcsuite.
+					client.
+					GetCSPICountEventually(cspc.Name, cspc.Namespace, 0)
+				Expect(gotCSPICount).To(BeNumerically("==", 0))
+			})
+
+			It("No corresponding pool-manger deployments should be present", func() {
+				gotPoolMangerCount := cspcsuite.
+					client.
+					GetPoolManagerCountEventually(cspc.Name, cspc.Namespace, 0)
+				Expect(gotPoolMangerCount).To(BeNumerically("==", 0))
+			})
+
+			It("the bdc(s) created by cstor-operator should get deleted", func() {
+				gotCount := cspcsuite.
+					client.
+					GetBDCCountEventually(cspc.Name, cspc.Namespace, 0)
+				Expect(gotCount).To(BeNumerically("==", 0))
+			})
```

Comment:
After this, we need to remove `UsedNodes` and `UsedDisks` from CSPCSpecData, right? Since they were no more using

Reply:
Done.